avm/res/compute/virtual-machine/main.bicep

```diff
@@ -522,7 +565,12 @@ resource vm 'Microsoft.Compute/virtualMachines@2022-11-01' = {
             primary: index == 0 ? true : false
           }
           #disable-next-line use-resource-id-functions // It's a reference from inside a loop which makes resolving it using a resource reference particulary difficult.
-          id: az.resourceId('Microsoft.Network/networkInterfaces', '${name}${nicConfiguration.nicSuffix}')
+          id: az.resourceId(
+            'Microsoft.Network/networkInterfaces',
+            contains(nicConfiguration, 'customNicName')
```

Comment:
I'd suggest to keep `name` instead of `customNicName` for the `nicConfiguration` property. Also, can we add this option to the max test?

Reply:
Done